src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultProjectWorkspaceStateGenerator.cs

```diff
@@ -56,7 +56,7 @@ public override void Initialize(ProjectSnapshotManagerBase projectManager)
             _tagHelperResolver = _projectManager.Workspace.Services.GetRequiredService<TagHelperResolver>();
         }
 
-        public override void Update(Project workspaceProject, ProjectSnapshot projectSnapshot)
+        public override void Update(Project workspaceProject, ProjectSnapshot projectSnapshot, CancellationTokenSource cts = default)
```

Comment:
Looks like these files shouldn't be affected by https://github.com/dotnet/aspnetcore/pull/21581/files

Reply:
This should take a `CancellationToken` instead of a `CancellationTokenSource`. Also, does it need to have a default or can we always provide one?

Reply:
I was hoping to keep `CancellationTokenSource` because of how we later store the `cts` (https://github.com/dotnet/aspnetcore-tooling/pull/1936/files#diff-8d2f7fc40b9c4a7c4aeb497a628bca44R96-R97) for later cancellation within this updateTask. I'm essentially "sharing" the CTS between the WorkspaceStateGenerator and the WorkspaceProjectStateChangeDetector Would it be acceptable in that case?

Reply:
You can still do that, instead you'll just created a linked token source: `CancellationTokenSource.CreateLinkedTokenSource(cancellationToken)` 😄

Reply:
Thanks, updated :)

Reply:
> Also, does it need to have a default or can we always provide one?

Updated to always provide `CancellationToken.None` at the very least (if another contextual cancellation token wasn't available).